The 5.7.0 feature "android: Make the ACCESS_WIFI_STATE permission optional" seems incomplete

<!-- NOTE: The iOS Simulator has some known issues with report network state changes. Please DO NOT report iOS bugs unless you are seeing them on a real device. -->

## Environment
<!-- Run `react-native info` in your terminal and paste its contents here. -->
info Fetching system and libraries information...
System:
    OS: Windows 10
  Binaries:
    Node: 10.17.0 - C:\Program Files\nodejs\node.EXE
    npm: 6.11.3 - C:\Program Files\nodejs\npm.CMD
  SDKs:
    Android SDK:
      API Levels: 27, 28, 29
      Build Tools: 28.0.3, 29.0.2
      System Images: android-27 | Google Play Intel x86 Atom, android-28 | Wear OS Intel x86 Atom, android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom, android-29 | Google APIs Intel x86 Atom
  IDEs:
    Android Studio: Version  3.5.0.0 AI-191.8026.42.35.5791312
  npmPackages:
    react: 16.9.0 => 16.9.0
    react-native: 0.61.4 => 0.61.4

## Platforms
<!-- Is this issue related to Android, iOS, or both? -->
Android

## Versions
<!-- Please add the used versions/branches -->
- Android: Api 28
- react-native-netinfo: 5.7.0
- react-native: 0.61.4
- react: 16.9.0

## Description
<!-- Describe your issue in detail. Include screenshots if needed. If this is a regression, let us know. -->

The 5.7.0 feature "Make the ACCESS_WIFI_STATE permission optional" seems incomplete.
This statement, line 110, in createConnectivityEventMap seems like it should be wrapped with the permission check:
`boolean isEnabled = mWifiManager.isWifiEnabled();`

Here is the exception:
```
2020-04-23 21:14:22.482 23733-23843 E/AndroidRuntime: FATAL EXCEPTION: ConnectivityThread
    java.lang.SecurityException: WifiService: Neither user 10270 nor current process has android.permission.ACCESS_WIFI_STATE.
        at android.os.Parcel.createException(Parcel.java:1966)
        at android.os.Parcel.readException(Parcel.java:1934)
        at android.os.Parcel.readException(Parcel.java:1884)
        at android.net.wifi.IWifiManager$Stub$Proxy.getWifiEnabledState(IWifiManager.java:1843)
        at android.net.wifi.WifiManager.getWifiState(WifiManager.java:2236)
        at android.net.wifi.WifiManager.isWifiEnabled(WifiManager.java:2248)
        at com.reactnativecommunity.netinfo.ConnectivityReceiver.createConnectivityEventMap(ConnectivityReceiver.java:116)
        at com.reactnativecommunity.netinfo.ConnectivityReceiver.sendConnectivityChangedEvent(ConnectivityReceiver.java:109)
        at com.reactnativecommunity.netinfo.ConnectivityReceiver.updateConnectivity(ConnectivityReceiver.java:102)
        at com.reactnativecommunity.netinfo.NetworkCallbackConnectivityReceiver.updateAndSend(NetworkCallbackConnectivityReceiver.java:112)
        at com.reactnativecommunity.netinfo.NetworkCallbackConnectivityReceiver$ConnectivityNetworkCallback.onAvailable(NetworkCallbackConnectivityReceiver.java:120)
        at android.net.ConnectivityManager$NetworkCallback.onAvailable(ConnectivityManager.java:3116)
        at android.net.ConnectivityManager$CallbackHandler.handleMessage(ConnectivityManager.java:3315)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:214)
        at android.os.HandlerThread.run(HandlerThread.java:65)
     Caused by: android.os.RemoteException: Remote stack trace:
        at android.app.ContextImpl.enforce(ContextImpl.java:1924)
        at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:1952)
        at com.android.server.wifi.WifiServiceImpl.enforceAccessPermission(WifiServiceImpl.java:1594)
        at com.android.server.wifi.WifiServiceImpl.getWifiEnabledState(WifiServiceImpl.java:1983)
        at android.net.wifi.IWifiManager$Stub.onTransact(IWifiManager.java:367)
```

## Reproducible Demo
<!-- Let us know how to reproduce the issue. Include a code sample or share a project that reproduces the issue. -->
<!-- Please follow the guidelines for providing a minimal example: https://stackoverflow.com/help/mcve -->
It is sufficient to:
`import NetInfo from "@react-native-community/netinfo";`
With this in the manifest file:
`    <uses-permission tools:node="remove" android:name="android.permission.ACCESS_WIFI_STATE" />
`

--- a/android/src/main/java/com/reactnativecommunity/netinfo/ConnectivityReceiver.java
+++ b/android/src/main/java/com/reactnativecommunity/netinfo/ConnectivityReceiver.java
@@ -6,15 +6,12 @@
  */
 package com.reactnativecommunity.netinfo;
 
-import android.Manifest;
 import android.content.Context;
-import android.content.pm.PackageManager;
 import android.net.ConnectivityManager;
 import android.net.wifi.WifiInfo;
 import android.net.wifi.WifiManager;
 import android.telephony.TelephonyManager;
 
-import androidx.core.content.ContextCompat;
 import androidx.core.net.ConnectivityManagerCompat;
 
 import com.facebook.react.bridge.Arguments;
@@ -115,8 +112,10 @@ abstract class ConnectivityReceiver {
         WritableMap event = Arguments.createMap();
 
         // Add if WiFi is ON or OFF
-        boolean isEnabled = mWifiManager.isWifiEnabled();
-        event.putBoolean("isWifiEnabled", isEnabled);
+        if (NetInfoUtils.isAccessWifiStatePermissionGranted(getReactContext())) {
+            boolean isEnabled = mWifiManager.isWifiEnabled();
+            event.putBoolean("isWifiEnabled", isEnabled);
+        }
 
         // Add the connection type information
         event.putString("type", requestedInterface != null ? requestedInterface : mConnectionType.label);
@@ -161,8 +160,7 @@ abstract class ConnectivityReceiver {
                 }
                 break;
             case "wifi":
-                if (ContextCompat.checkSelfPermission(getReactContext(),
-                        Manifest.permission.ACCESS_WIFI_STATE) == PackageManager.PERMISSION_GRANTED) {
+                if (NetInfoUtils.isAccessWifiStatePermissionGranted(getReactContext())) {
                     WifiInfo wifiInfo = mWifiManager.getConnectionInfo();
                     if (wifiInfo != null) {
                         // Get the SSID

--- a/android/src/main/java/com/reactnativecommunity/netinfo/NetInfoUtils.java
+++ b/android/src/main/java/com/reactnativecommunity/netinfo/NetInfoUtils.java
@@ -6,6 +6,12 @@
  */
 package com.reactnativecommunity.netinfo;
 
+import android.Manifest;
+import android.content.Context;
+import android.content.pm.PackageManager;
+
+import androidx.core.content.ContextCompat;
+
 public class NetInfoUtils {
     public static void reverseByteArray(byte[] array) {
         for (int i = 0; i < array.length / 2; i++) {
@@ -13,5 +19,10 @@ public class NetInfoUtils {
             array[i] = array[array.length - i - 1];
             array[array.length - i - 1] = temp;
         }
+    }
+
+    public static boolean isAccessWifiStatePermissionGranted(Context context) {
+        return ContextCompat.checkSelfPermission(context,
+                Manifest.permission.ACCESS_WIFI_STATE) == PackageManager.PERMISSION_GRANTED;
     }
 }

